p2p/node/localnode.go

```diff
@@ -35,15 +35,15 @@ func NewNodeIdentity() (LocalNode, error) {
 }
 
 // Creates a new node from persisted NodeData.
-func newLocalNodeFromFile(d *nodeFileData) (LocalNode, error) {
+func newLocalNodeFromFile(d *nodeFileData) (ln LocalNode, err error) {
 	priv, err := p2pcrypto.NewPrivateKeyFromBase58(d.PrivKey)
 	if err != nil {
-		return emptyNode, err
+		return ln, fmt.Errorf("failed to parse private key: %w", err)
 	}
 
 	pub, err := p2pcrypto.NewPublicKeyFromBase58(d.PubKey)
 	if err != nil {
-		return emptyNode, err
+		return ln, fmt.Errorf("failed to parse public key: %w", err)
```

Comment:
Add context to these errors, so that we can tell when a failure happens from parsing each type of key.